lib/scoring_bowling.rb

```diff
@@ -0,0 +1,67 @@
+class ScoringBowling
+  MAX_FRAMES = 10
+  MAX_PINS = 10
+  attr_reader :rounds, :frame, :score
+
+  def initialize
+    @rounds = []
+    @frame = 0
+    @score = 0
+  end
+
+  def add_bowl(*bowls)
+    raise 'Game Over, no more scores can be added' if @rounds.length == MAX_FRAMES
+    
+    @rounds << { frame: frame_count, bowls: [*bowls] }
+  end
+  
+  def calculate_score
+    current_frame = 0
+    arr = []
+    # Needs to be refactored, unsure how.
+    until current_frame == rounds.length
+      calculate_all_scenarios(current_frame, arr)
+      current_frame += 1
+    end
+    @score += arr.flatten.sum
+  end
+
+  private
+
+  def calculate_all_scenarios(which_frame, arr)
+    if strike?(round(which_frame))
+      calculate_strike(which_frame, arr)
+    elsif spare?(round(which_frame))
+      calculate_spare(which_frame, arr)
+    else
+      arr << round(which_frame)
```

Comment:
Can we access this array from here?

Reply:
poorley named parameter. I should make it clear what it is.